### PR TITLE
chore(medusa): Add transaction on mutation actions store

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Wrap all actions that require the usage of an atomic phase into a transaction from the handler

Fixes CORE-358

### CODE REVIEW

File: packages/medusa/src/api/routes/store/carts/calculate-taxes.ts [hunk 1]
```diff
@@ -34,91 +34,98 @@ export default async (req, res) => {
...
+      idempotencyKey = await idempotencyKeyService.withTransaction(transactionManager).initializeRequest(
+        headerKey,
+        req.method,
+        req.params,
+        req.path
+      )
+    } catch (error) {
+      console.log(error)
+      res.status(409).send("Failed to create idempotency key")
+      return
```
Comment (srindom) [inline packages/medusa/src/api/routes/store/carts/calculate-taxes.ts:50]: todo: the initialization of the idempotency key must happen within a transaction otherwise the subsequent calls to `idempotencyKeyService.workStage` will create nested transactions which I don't think will work.

Comment (adrien2p) [reply packages/medusa/src/api/routes/store/carts/calculate-taxes.ts:50]: @srindom I did pass the transaction, what do you mean?

**Review by pKorsholm — COMMENTED** [review]
With Sebs comment resolved I think it looks good 😄

**Review by olivermrbl — DISMISSED** [review]
LGTM!

File: packages/medusa/src/api/routes/store/carts/update-cart.ts [hunk 2]
```diff
@@ -89,15 +90,19 @@ export default async (req, res) => {
...
-  await cartService.update(id, validated)
+  const manager: EntityManager = req.scope.resolve("manager")
 
-  const updated = await cartService.retrieve(id, {
-    relations: ["payment_sessions", "shipping_methods"],
-  })
+  await manager.transaction(async (transactionManager) => {
+    await cartService.withTransaction(transactionManager).update(id, validated)
+
+    const updated = await cartService.retrieve(id, {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/store/carts/update-cart.ts:98]: question: Why did you remove the use of `withTransaction` here?

Comment (adrien2p) [reply packages/medusa/src/api/routes/store/carts/update-cart.ts:98]: Conflict not that well managed 🤘

**Review by olivermrbl — APPROVED** [review]
LGTM!
